### ISSUE TITLE
Android / OUYA compatibility

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#464]
*Edit 2017-2-9: This item is blocked by prerequisites like #2547 and is not a viable GSOC item until that topic makes more progress*

Issue to go with the Incubator thread at https://forum.terasology.org/threads/android-ouya.679/ - much more information there :-)

I've got the expensive pre-release developer special with a year of promotion and such, which I'd like to see used - but I'm admitting it is a fair undertaking and normal OUYA release is already close. Hopefully we'll be done before the freebie year is over ... ;-)

If somebody were to show substantial progress on this item (likely via a plain Android version) I'd be happy to help get said person set up with an actual OUYA and release (even unrelated) games with the shiny "founder" account I get along with the special benefits it comes with - if that helps sweeten the pot!
##

**Comment by Member1221** [comment]
My laptop is a tablet hybrid (touch screen, etc.) But afaik, windows applications for the tablet mode needs to be made in C++, C#, etc? I don't know if there is someone who have made a java compatible framework for it.

**Comment by Cervator** [comment]
You mean the Windows "Metro" mode or whatever it is called? That might be, but you don't need that. I've since run Terasology on my Surface Pro and the touch screen works fine, although it is just simulating the mouse which isn't exactly a pleasant experience.

Destination Sol is in Java and works perfectly on an Android touch screen. I don't really care about Windows tablet mode, although there might be some worth there to look into at some point.

Adopting LibGDX is likely another new plan that has come about in the two years since I posted this. I'm sure there are some touch screen utilities in there since DestSol is working that way.

Pretty advanced/complex topic in any case.

**Comment by rodneyrod** [comment]
Is it compiling by any chance at least? Might be able to get it working with a controller or keyboard+ mouse while a touchscreen UI is worked out?

**Comment by skaldarnar** [comment]
This issue is really ancient by now 🙄  I'm not sure the AWT facade is still working (I suspect it is not). We've done no progress on touchscreen UI (and this is not a focus topic right now). We did move the rendering DAG ouf the engine to https://github.com/Terasology/CoreRendering and upgrade to LWJGL v3 (see https://github.com/MovingBlocks/Terasology/pull/3969). That left us with a bunch of follow-ups to work on (see https://github.com/MovingBlocks/Terasology/issues/4206).

At this point, I don't know what this issue is supposed to be about, to be honest... @rodneyrod is there anything in particular you would be interested in? Controllers support could use some improvements (see https://github.com/MovingBlocks/Terasology/pull/4218#issuecomment-720076283), for instance. Not sure who can help with Android support - maybe @DarkWeird knows more on how this relates to the LWJGL and gestalt v7 workstreams...

**Comment by DarkWeird** [comment]
Yeah AWT not work now. (LWJGL3). needs to implement FacadeAWT over `org.lwjglx.jawt` if awt needs.

I began splitting subsystems, then we can try to implement touchscreen and  graphics for android.
Also we needs gestaltv7 (current v5 in Terasology) - it works with android.

And it all will works, while we don't write code for java 9+ in engine.
